### PR TITLE
decorate a class

### PR DESCRIPTION
**Pull request by PrinceBadru**
## Description
In this project, you will use the Decorator design pattern to correct the names of people.
 - Check the usage of [Decorator pattern in Ruby](https://refactoring.guru/design-patterns/decorator/ruby/example#example-0)
 - Think about how you can use two decorators in order to capitalize and trim people's names.
 - In the following steps you will get instructions for implementing the Decorator design pattern for this case.
 - Create a class Nameable
 - Turn your Person class to Nameable
 - Prepare base Decorator
 - Prepare CapitalizeDecorator and TrimmerDecorator

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,8 +1,11 @@
-class Person
+require_relative 'nameable'
+
+class Person < Nameable
   attr_accessor :name, :age
   attr_reader :id
 
   def initialize(age, name: 'Unknown', parent_permission: true)
+    super() # this line to calls the initialize method of the parent class
```
Comment (zeff96) [inline person.rb:8]: - [ ] Hello, @PrinceBadru! 👋 Fantastic job on your progress so far! According to the project requirements, we need you to create a method called correct_name that will retrieve and return the value of the name attribute. 🚀 Let's make sure this method smoothly and accurately handles the task at hand. 💡

```ruby
 def correct_name
       name
 end
```

**Review by zeff96 — CHANGES_REQUESTED** [review]
### Status: CHANGES REQUIRED ♻️
Hi @PrinceBadru , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
:heavy_check_mark: Linters passing
:heavy_check_mark: Correct GitFlow



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by danmainah — APPROVED** [review]
Hi @PrinceBadru , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️  Implemented the change request to ensure the correct implementation of  `current_name`  method in the person class.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by PrinceBadru** [comment]
@danmainah @zeff96  thanks guys

**Pull request merged**